lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -312,28 +311,18 @@ _estimateMaxNumberOfInputs
     -- ^ Number of outputs in transaction
     -> Word8
 _estimateMaxNumberOfInputs networkId (Quantity maxSize) md nOuts =
-      fromIntegral $ bisect (lowerBound, upperBound)
+    findMax minBound
   where
-    bisect (!inf, !sup)
-        | middle == inf && isTooBig sup = inf
-        | middle == inf                 = sup
-        | isTooBig middle               = bisect (inf, middle)
-        | otherwise                     = bisect (middle, sup)
-      where
-        middle = inf + ((sup - inf) `div` 2)
-
-    growingFactor = 2
-
-    lowerBound = upperBound `div` growingFactor
-    upperBound = upperBound_ 1
-      where
-        upperBound_ !n | isTooBig n = n
-                       | otherwise  = upperBound_ (n*growingFactor)
+    findMax :: Word8 -> Word8
+    findMax inf
+        | inf == maxBound    = 0
+        | isTooBig (inf + 1) = inf
+        | otherwise          = findMax (inf + 1)
 
     isTooBig nInps = size > fromIntegral maxSize
```

Comment:
What would you think about
1. Calling this `txTooBig`
2. Passing it explicitly to the `findMax` function

Reply:
1. Sounds good.
2. Okay, but why ? What's the added benefit in this context?

Reply:
I think `findMax :: Word8 -> Word8` is a kind-of deceptive name and type-signature. It sounds abstract, but it is not. I believe it is meant more as a helper of `estimateMaxNumberOfInputs`.

I think it should either _be_ abstract, _or_ have a more specialised name.

I think explicitly showing the dependency on the `txTooBig` condition might be clearer in both cases, but in particular, it _would_ make `findMax` abstract (assuming no other details specific to this use-case).

Modulo, if the type-signature becomes too ugly, perhaps.